### DISCUSSION TITLE
Set default calendar when adding an event

### DISCUSSION THREAD

**Discussion by darac** [home-assistant/frontend#19291] [Other feature requests]
If multiple calendars can be written to, when the user presses "Add Event" a dialog appears to gather the information for the new calendar event. This dialog has a drop-down that allows the user to choose which calendar the event ends up being written to.

Several times I have created a calendar event and forgotten to change the drop-down, so the event has ended up on the wrong calendar. (And, of course, you can't move entries between calendars).

I can think of a few things that might help here (implementing all of these doesn't make sense, these are just some options):
1. Allow for setting the default calendar somewhere. This might be somewhere in a YAML file, or there might be a configuration control somewhere. The "Add Event" calendar would then respect this by defaulting the dropdown to the chosen calendar (if it exists).
2. Remember the last-chosen calendar. When a user adds a calendar event, it seems reasonable to assume that the next event they want to create _might_ want to be in the same calendar. Remembering which calendar was chosen and simply defaulting the dialog to that calendar might make things better.

**Comment by kfiducia** [comment]
Both good ideas - I am wondering if the intent was never to use local calendars and always to manage calendars elsewhere and subscribe to them. As well done as the calendars features is here, it also has a couple missing features that could make it much more functional (like the above). Also - maybe a an ability to add actions to the start/end. I know you can do this a million ways, like automations, but for users that have LOTS of automations, calendars offer a nice way to see what is automatically happening throughout the day and when.

**Comment by GH2user** [comment]
Yes, good idea.
Even when unselecting all calendars but one it still doesn't pre-select the one shown which would already be a big plus for me.

**Comment by gorischek** [comment]
I would prefer solution 1, but both would be ok. And very helpful :-)

**Comment by msknight** [comment]
I would prefer for option 1. I use four local calendars, one for me, one for my mother, one for medical, etc... so that I can change the colour in the calendar display so I can see at a glance what the event is for... and frequently I'm ending up creating an event in the wrong calendar by accident and then having to destroy the event and re-create it from scratch because there is no option to change calendars once an event is created. If I could set a default calendar then I could at least develop a reflex action to change the calendar.

**Comment by reineo** [comment]
Any update on this proposal? Having automation calendars as the default isn't very useful and causes a lot of problems day to day. I would prefer if I can select what calendar should be default when adding new events. I.e my external Google calendar.
